rs/registry/helpers/src/routing_table.rs

```diff
@@ -24,16 +24,24 @@ pub trait RoutingTableRegistry {
 
 impl<T: RegistryClient + ?Sized> RoutingTableRegistry for T {
     fn get_routing_table(&self, version: RegistryVersion) -> RegistryClientResult<RoutingTable> {
-        let bytes = self.get_value(&make_routing_table_record_key(), version);
-        deserialize_registry_value::<pb::RoutingTable>(bytes).map(|option_pb_routing_table| {
-            option_pb_routing_table
-                .map(|pb_routing_table| {
-                    RoutingTable::try_from(pb_routing_table).map_err(|err| DecodeError {
-                        error: format!("get_routing_table() failed with {}", err),
-                    })
+        let canister_ranges_keys = self.get_key_family(CANISTER_RANGES_PREFIX, version)?;
+        let routing_table_shards = canister_ranges_keys
+            .iter()
+            .map(|key| {
+                let bytes = self.get_value(key, version);
+                deserialize_registry_value::<pb::RoutingTable>(bytes)?.ok_or_else(|| DecodeError {
+                    error: format!(
+                        "canister ranges key {} does not have a routing table shard",
+                        key
+                    ),
                 })
-                .transpose()
-        })?
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        RoutingTable::try_from(routing_table_shards)
+            .map(Some)
```

Comment:
The behavior changed for the case of no routing table key (no canister ranges shards):
- before this PR, the return value would be `Ok(None)`;
- after this PR, the return value would be `Ok(Some(...))` with an _empty_ routing table.

That's probably fine because registry invariants (currently) ensure that there must be a routing table key in the registry. But if the corresponding [check](https://github.com/dfinity/ic/blob/e915efecc8af90993ccfc499721ebe826aadba60/rs/registry/canister/src/invariants/routing_table.rs#L25-L34) is removed, then this might be a breaking change.

Reply:
I adjusted it so that if the keys comes back empty, it returns Ok(None) instead.

Reply:
(Please resolve this conversation if that satisfies you)

Reply:
LGTM